service/share/discovery.go

```diff
@@ -87,13 +106,25 @@ func (d *discovery) findPeers(ctx context.Context) {
 			for peer := range peers {
 				go d.handlePeerFound(ctx, topic, peer)
 			}
+		case e := <-sub.Out():
+			// listen to disconnect event to remove peer from set and reset backoff time
+			// reset timer in order to restart the discovery, once stored peer is disconnected
+			connStatus := e.(event.EvtPeerConnectednessChanged)
+			if connStatus.Connectedness != network.NotConnected || !d.set.Contains(connStatus.Peer) {
+				continue
+			}
+			d.connector.RestartBackOff(connStatus.Peer)
+			d.set.Remove(connStatus.Peer)
+			d.host.ConnManager().UntagPeer(connStatus.Peer, topic)
+			t.Reset(interval * 3)
```

Comment:
maybe extract this in something called `defaultBackoffInterval` or something ? since you use this val twice